sklearn/metrics/classification.py

```diff
@@ -391,72 +392,190 @@ def jaccard_similarity_score(y_true, y_pred, normalize=True,
     y_pred : 1d array-like, or label indicator array / sparse matrix
         Predicted labels, as returned by a classifier.
 
-    normalize : bool, optional (default=True)
-        If ``False``, return the sum of the Jaccard similarity coefficient
-        over the sample set. Otherwise, return the average of Jaccard
-        similarity coefficient.
+    labels : list, optional
+        The set of labels to include when ``average != 'binary'``, and their
+        order if ``average is None``. Labels present in the data can be
+        excluded, for example to calculate a multiclass average ignoring a
+        majority negative class, while labels not present in the data will
+        result in 0 components in a macro average. For multilabel targets,
+        labels are column indices. By default, all labels in ``y_true`` and
+        ``y_pred`` are used in sorted order.
+
+    pos_label : str or int, 1 by default
+        The class to report if ``average='binary'`` and the data is binary.
+        If the data are multiclass or multilabel, this will be ignored;
+        setting ``labels=[pos_label]`` and ``average != 'binary'`` will report
+        scores for that label only.
+
+    average : string, [None (default), 'binary', 'micro', 'macro', 'samples', \
+                       'weighted']
+        If ``None``, the scores for each class are returned. Otherwise, this
+        determines the type of averaging performed on the data:
+
+        ``'binary'``:
+            Only report results for the class specified by ``pos_label``.
+            This is applicable only if targets (``y_{true,pred}``) are binary.
+        ``'micro'``:
+            Calculate metrics globally by counting the total true positives,
+            false negatives and false positives.
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean.  This does not take label imbalance into account.
+        ``'weighted'``:
+            Calculate metrics for each label, and find their average, weighted
+            by support (the number of true instances for each label). This
+            alters 'macro' to account for label imbalance.
+        ``'samples'``:
+            Calculate metrics for each instance, and find their average (only
+            meaningful for multilabel classification).
+        ``'none-samples'``:
+
+    warn : bool, optional (default=True), for internal use
```

Comment:
I intended this `warn` to be similar to `warn_for` in `precision_recall_fscore_support`, see https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/metrics/classification.py#L881 , though it uses `warn_for`, but since this is a single metric (P/R/F/S are 4), so I thought to instead simply use `warn` and use a boolean value for it.

Should we use a different name maybe, if this creates a confusion with `warnings.warn`?

Reply:
But that's only really in prf for internal use. No need to expose it publicly.

Reply:
> No need to expose it publicly.

I'm sorry, I still don't understand, how have I exposed here `warn` publicly?

If you mean writing about `warn` here in the docstring, then the same is done for `warn_for` in P/R/F here https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/metrics/classification.py#L960 .

I don't think I've made `warn` different from `warn_for`.